tools/pubsys/src/aws/ami/launch_permissions.rs

```diff
@@ -0,0 +1,97 @@
+use aws_sdk_ec2::{model::LaunchPermission, Client as Ec2Client};
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+
+/// Returns the launch permissions for the given AMI
+pub(crate) async fn get_launch_permissions(
+    ec2_client: &Ec2Client,
+    region: &str,
+    ami_id: &str,
+) -> Result<Vec<LaunchPermissionDef>> {
+    let ec2_response = ec2_client
+        .describe_image_attribute()
+        .image_id(ami_id)
+        .attribute(aws_sdk_ec2::model::ImageAttributeName::LaunchPermission)
+        .send()
+        .await
+        .context(error::DescribeImageAttributeSnafu {
+            ami_id,
+            region: region.to_string(),
+        })?;
+
+    let mut launch_permissions = vec![];
+
+    let responses: Vec<LaunchPermission> =
+        ec2_response.launch_permissions().unwrap_or(&[]).to_vec();
+    for permission in responses {
+        launch_permissions.push(LaunchPermissionDef::try_from(permission)?)
+    }
+    Ok(launch_permissions)
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq, Hash)]
+pub(crate) enum LaunchPermissionDef {
```

Comment:
I think it would be more coherent with our other serialization code if you lowercased these:

```suggestion
#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq, Hash)]
#[serde(rename_all = "lowercase")]
pub(crate) enum LaunchPermissionDef {
```

Reply:
Wouldn't it make more sense to use snake_case?

Reply:
If I `git grep` for `rename_all` in Bottlerocket, it seems like we typically use `kebab-case` or `lowercase`